tests/renderToString.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { defineComponent, onMounted, onServerPrefetch, ref } from 'vue'
+import { renderToString } from '../src'
+
+describe('renderToString', () => {
+  it('returns a promise', async () => {
+    const Component = defineComponent({
+      template: '<div>{{ text }}</div>',
+      setup() {
+        return { text: 'Text content' }
+      }
+    })
+
+    const wrapper = await renderToString(Component)
+
+    expect(wrapper).toMatchInlineSnapshot(`"<div>Text content</div>"`)
+  })
+
+  it('returns correct html on multi root nodes', async () => {
+    const Component = defineComponent({
+      template: '<div>foo</div><div>bar</div>'
+    })
+
+    const wrapper = await renderToString(Component)
+
+    expect(wrapper).toMatchInlineSnapshot(
+      `"<!--[--><div>foo</div><div>bar</div><!--]-->"`
+    )
+  })
+
+  it('returns correct html when onServerPrefetch is used', async () => {
+    const Component = defineComponent({
+      template: '<div>{{ text }}</div>',
+      setup() {
+        const text = ref('')
+
+        onServerPrefetch(() => {
+          return new Promise((resolve) => {
+            setTimeout(() => {
+              text.value = 'Text content'
+              resolve(true)
+            }, 100)
+          })
+        })
```

Comment:
onServerPrefetch sets the text to `Text content`, and `onMounted` as well: which one is tested in this test? Maybe they should have different values, and we should check both in the test?

Reply:
Maybe I can update this part to check if the rendered text is the text from pre-fetched data? Something like

```ts
it('returns correct html with pre-fetched data on server', async () => {
  function fakeFetch(text: string) {
    return new Promise<string>((resolve) => {
      setTimeout(() => {
        resolve(text)
      }, 100)
    })
  }

  const Component = defineComponent({
    template: '<div>{{ text }}</div>',
    setup() {
      const text = ref<string | null>(null)

      onServerPrefetch(async () => {
        text.value = await fakeFetch('onServerPrefetch')
      })

      onMounted(async () => {
        if (!text.value) {
          text.value = await fakeFetch('onMounted')
        }
      })

      return { text }
    }
  })

  const contents = await renderToString(Component)

  expect(contents).toBe('<div>onServerPrefetch</div>')
})
```

Basing on this one https://vuejs.org/api/composition-api-lifecycle.html#onserverprefetch

Reply:
Yes I think that would be great 👍

Reply:
updated 👍🏼